packages/mdc-floating-label/foundation.ts

```diff
@@ -21,67 +21,59 @@
  * THE SOFTWARE.
  */
 
+import {SpecificEventListener} from '@material/base';
 import {MDCFoundation} from '@material/base/foundation';
-import MDCFloatingLabelAdapter from './adapter';
+import {MDCFloatingLabelAdapter} from './adapter';
 import {cssClasses} from './constants';
 
-/**
- * @extends {MDCFoundation<!MDCFloatingLabelAdapter>}
- * @final
- */
-class MDCFloatingLabelFoundation extends MDCFoundation {
-  /** @return enum {string} */
+class MDCFloatingLabelFoundation extends MDCFoundation<MDCFloatingLabelAdapter> {
   static get cssClasses() {
     return cssClasses;
   }
 
   /**
-   * {@see MDCFloatingLabelAdapter} for typing information on parameters and return
-   * types.
-   * @return {!MDCFloatingLabelAdapter}
+   * See {@link MDCFloatingLabelAdapter} for typing information on parameters and return types.
    */
-  static get defaultAdapter() {
-    return /** @type {!MDCFloatingLabelAdapter} */ ({
-      addClass: () => {},
-      removeClass: () => {},
-      getWidth: () => {},
-      registerInteractionHandler: () => {},
-      deregisterInteractionHandler: () => {},
-    });
+  static get defaultAdapter(): MDCFloatingLabelAdapter {
+    // tslint:disable:object-literal-sort-keys
+    return {
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      getWidth: () => 0,
+      registerInteractionHandler: () => undefined,
+      deregisterInteractionHandler: () => undefined,
+    };
+    // tslint:enable:object-literal-sort-keys
   }
 
-  /**
-   * @param {!MDCFloatingLabelAdapter} adapter
-   */
-  constructor(adapter) {
+  private readonly shakeAnimationEndHandler_: SpecificEventListener<'animationend'>;
+
+  constructor(adapter: MDCFloatingLabelAdapter) {
     super(Object.assign(MDCFloatingLabelFoundation.defaultAdapter, adapter));
 
-    /** @private {function(!Event): undefined} */
     this.shakeAnimationEndHandler_ = () => this.handleShakeAnimationEnd_();
   }
 
   init() {
-    this.adapter_.registerInteractionHandler('animationend', this.shakeAnimationEndHandler_);
+    this.adapter_.registerInteractionHandler('animationend', this.shakeAnimationEndHandler_ as EventListener);
   }
 
   destroy() {
-    this.adapter_.deregisterInteractionHandler('animationend', this.shakeAnimationEndHandler_);
+    this.adapter_.deregisterInteractionHandler('animationend', this.shakeAnimationEndHandler_ as EventListener);
   }
 
   /**
    * Returns the width of the label element.
-   * @return {number}
    */
-  getWidth() {
+  getWidth(): number {
     return this.adapter_.getWidth();
   }
 
   /**
-   * Styles the label to produce the label shake for errors.
-   * @param {boolean} shouldShake adds shake class if true,
-   * otherwise removes shake class.
+   * Styles the label to produce a shake animation to indicate an error.
+   * @param shouldShake If true, adds the shake CSS class; otherwise, removes shake class.
    */
-  shake(shouldShake) {
+  shake(shouldShake: boolean): void {
```

Comment:
The `: void` returns are unnecessary, right?

Reply:
correct - I think they are inferred

Reply:
Removed